src/dftbp/timedep/timeprop.F90

```diff
@@ -4107,7 +4117,7 @@ subroutine doTdStep(this, boundaryCond, iStep, coord, orb, neighbourList, nNeigh
         velInternal(:,:) = 0.0_dp
       end if
       call writeRestartFile(this%rho, this%rhoOld, coord, velInternal, this%time, this%dt,&
-          & trim(dumpIdx) // 'ppdump', this%tWriteRestartAscii, errStatus)
+          & trim(pumpFilesDir) // '/' // trim(dumpIdx) // 'ppdump', this%tWriteRestartAscii, errStatus)
```

Comment:
Joining file paths should account for the operating system.

Reply:
Agreed, would using this procedure be fine?
```
CALL GET_ENVIRONMENT_VARIABLE('PATH',path)
path_separator=path(1:1)
```

Reply:
If your `PATH` starts with a drive letter `C:\` this will not work.

Reply:
The correct solution would be IMO a pre-processor based approach, setting the separator based on the system, as reported in the `${_SYSTEM_}$` variable in Fypp. (And actually a function, like `joinPaths()` which hides all those nasty details. Or even better, a Path class, like in recent python releases....)